include/runtime/format.h

```diff
@@ -0,0 +1,66 @@
+#ifndef RUNTIME_FORMAT_H
```

Comment:
I believe boost has a version of this library that the c++20 library is based on. We probably want to use it rather than rolling our own implementation.

Reply:
As far as I can tell, the [boost library](https://www.boost.org/doc/libs/1_77_0/libs/format/doc/format.html) has a different API to `std::format`. I can drop in [`fmt`](https://github.com/fmtlib/fmt) easily, which is compatible (and the current de facto formatting library for C++).